### PR TITLE
add docs for cpp tests requiring built libraries

### PR DESCRIPTION
**Pull request by williamberman**
**Description**
Update contributor guidelines to note cpp tests require linking to built libraries (i.e. ONNXIFI_DUMMY_LIBRARY). 

**Motivation and Context**
- The cpp tests fail when running locally for me.
- The ci scripts also set LD_LIBRARY_PATH

### CODE REVIEW

File: docs/CONTRIBUTING.md [hunk 1]
```diff
@@ -86,7 +86,21 @@ python onnx\backend\test\stat_coverage.py
...
+```bash
+export LD_LIBRARY_PATH="./.setuptools-cmake-build/:$LD_LIBRARY_PATH
+.setuptools-cmake-build/onnx_gtests
+```
+
+### Windows
+```bat
+set PATH=./.setuptools-cmake-build/;%PATH%
+.setuptools-cmake-build/onnx_gtests
+```
```
Comment (williamberman) [inline docs/CONTRIBUTING.md:103]: Not sure if this is right for windows. Don't have a local windows dev environment set up to test

Comment (jcwchen) [reply docs/CONTRIBUTING.md:103]: From my local Windows, I can run `.setuptools-cmake-build\Debug\onnx_gtests.exe` directly without setting additional PATH.

File: docs/CONTRIBUTING.md [hunk 2]
```diff
@@ -86,7 +86,21 @@ python onnx\backend\test\stat_coverage.py
...
+### Linux and MacOS
+```bash
+export LD_LIBRARY_PATH="./.setuptools-cmake-build/:$LD_LIBRARY_PATH
+.setuptools-cmake-build/onnx_gtests
+```
+
+### Windows
+```bat
+set PATH=./.setuptools-cmake-build/;%PATH%
+.setuptools-cmake-build/onnx_gtests
```
Comment (jcwchen) [inline docs/CONTRIBUTING.md:102]:
```suggestion
# If you set DEBUG=1, use `.setuptools-cmake-build\Debug\onnx_gtests.exe` instead
.setuptools-cmake-build\Release\onnx_gtests.exe
```

**Review by jcwchen — COMMENTED** [review]
Comments about Windows part. The part of Linux and MacOS LGTM. Thanks for the improvement !

Please follow [the steps](https://github.com/onnx/onnx/blob/main/CONTRIBUTING.md#dco) to signoff DCO.

File: docs/CONTRIBUTING.md (hunk 2 above)
Comment (williamberman) [reply docs/CONTRIBUTING.md]: Added!

File: docs/CONTRIBUTING.md [hunk 3]
```diff
@@ -86,7 +86,21 @@ python onnx\backend\test\stat_coverage.py
...
 Some functionalities are tested with googletest. Those tests are listed in `test/cpp`, and include tests for shape inference, data propagation, parser, and others.
 
-To run them, first build ONNX with `-DONNX_BUILD_TESTS=1` or `ONNX_BUILD_TESTS=1 pip install -e .` and then run `.setuptools-cmake-build/onnx_gtests`.
+To run them, first build ONNX with `-DONNX_BUILD_TESTS=1` or `ONNX_BUILD_TESTS=1 pip install -e .`.
+
+### Linux and MacOS
+The cpp tests require dynamically linking to built libraries.
+
+```bash
+export LD_LIBRARY_PATH="./.setuptools-cmake-build/:$LD_LIBRARY_PATH
```
Comment (jcwchen) [inline docs/CONTRIBUTING.md:95]:
```suggestion
export LD_LIBRARY_PATH="./.setuptools-cmake-build/:$LD_LIBRARY_PATH"
```

**Review by jcwchen — APPROVED** [review]
LGTM. Thanks!

**Pull request merged**